NewareNDA/NewareNDA.py

```diff
@@ -270,7 +290,8 @@ def _generate_cycle_number(df, cycle_mode='chg'):
             _, cycle_mode = first_state.split('_', 1)
         except ValueError:
             # Status is SIM or otherwise. Set mode to chg
-            warnings.warn("First Step not recognized. Defaulting to Cycle_Mode 'Charge'.")
+            warnings.warn(f"First Step '{first_state}' not recognized. Defaulting to Cycle_Mode 'Charge'.")
+            logger.warning(f"First Step '{first_state}' not recognized. Defaulting to Cycle_Mode 'Charge'.")

```

Comment:
Will this mean that message is printed twice, or how does warnings.warn vs logging.warning work?

Reply:
My understanding is that `warnings.warn` will print to the console, but not the log file, and `logger.warning` will append the message to the log file, but will not print to console. I suppose I could have made a short function that would call both and use that, but at the moment it doesn't save us that many lines.

Reply:
According to the [Python logging documentation](https://docs.python.org/3/howto/logging.html), `warnings.warn` should only be used "if the issue is avoidable and the client application should be modified to eliminate the warning". I think we should be using `logging.warning()` in most cases here.